doc/developer_guide/developer_guide.md

```diff
@@ -0,0 +1,121 @@
+# Developer Guide
+
+## Preparation
+
+Before you can build and test the application, you need to install Lua packages on the build machine.
+
+### Installing LuaRocks
+
+First install the package manager LuaRocks.
+
+```bash
+sudo apt install luarocks
+```
+
+Now update your `LUA_PATH`, so that it contains the packages (aka. "rocks"). You can auto-generate that path.
+
+```bash
+luarocks path
+```
+
+Of course generating it is not enough, you also need to make sure the export is actually executed &mdash; preferably automatically each time. Here is an example that appends the path to the `.bashrc`:
+
+```bash
+luarocks path >> ~/.bashrc
+```
+
+### Installing the Required Lua Packages
+
+You need the packages for unit testing, mocking and JSON processing.
+
+Execute as `root` or modify to install in your home directory:
+
+```bash
+sudo luarocks install LuaUnit
+sudo luarocks install Mockagne
+sudo luarocks install lua-cjson
+sudo luarocks install remotelog
+sudo luarocks install luacov
+sudo luarocks install luacov-coveralls
+sudo luarocks install luacheck
+```
```

Comment:
Use

```
luarocks install --local --deps-only *.rockspec
```
if possible

Reply:
Done. I might have to distinguish between developer tools and real dependencies later (#5).